plutus-tx-plugin/test/Plugin/Spec.hs

```diff
@@ -387,3 +414,20 @@ valueRestriction = plc @"valueRestriction" (let { f :: forall a . a; f = Builtin
 
 recordSelector :: CompiledCode (MyMonoRecord -> Int)
 recordSelector = plc @"recordSelector" (\(x :: MyMonoRecord) -> mrA x)
+
+emptyRoseId1 :: CompiledCode (EmptyRose -> EmptyRose)
+emptyRoseId1 = plc @"emptyRoseId1" (
+    let map _ []     = []
+        map f (x:xs) = f x : map f xs
+        go (EmptyRose xs) = EmptyRose (map go xs)
+    in go)
```

Comment:
Did you try adding type signatures to see if that fixes it? At the moment you've checked in the error result.

Reply:
> Did you try adding type signatures to see if that fixes it? 

`sameEmptyRose` is very similar, but due to explicit types it does not result in the error.

> At the moment you've checked in the error result.

Yes, that's the point, the golden test is `errors/emptyRoseId1.plc.golden`. I think, it's important to have such a test, because it is not obvious that programs like this can violate the value restriction.

Reply:
Ah sorry, I didn't see that this was an error test.